tests/unit/rest.py

```diff
@@ -287,15 +294,15 @@ def _do_request(client, method, path, data):
     `data` should be a string -- the server will expect valid json, but
     we want to write test cases with invalid input as well.
     """
-    client.post('/give-me-an-e', data=data)
-    # Annoyingly, the client doesn't give us a way to supply an arbitrary
-    # HTTP method, so we hack around it by using getattr to get the right
-    # function:
-    return getattr(client, method.lower())(path, data=data)
+    if method == "GET":
+        return client.get(path, query_string=query, data=data)
+    else:
+        return client.open(method=method, path=path, data=data)
```

Comment:
Didn't know about this one, good find. Questions:

1. Can we just not have the if/else and pass the (possibly empty) query string to both?
2. If not, could we at least be consistent about passing path by name vs. position?